### ISSUE TITLE
Importing header/source files with same names and different paths

### ISSUE THREAD

**Issue by vojerr** [CocoaPods/CocoaPods#8289]
Cocoapods version: 1.5.3
Xcode version: 10.0

Consider following structure of local pod dependency with objective-c sources:
dependency -> folder1 -> MyClass.h
                      -> folder1 -> MyClass.m
                      -> folder2 -> MyClass.h
                      -> folder2 -> MyClass.h

Files MyClass in folder1 contains implementation of class, named Class1
Files MyClass in folder2 contains implementation of class, named Class2
It's general case, while using j2objc, that class names are different to files(different packages in java with same class names).

After importing this local dependency to cocoa pods workspace, files from only one folder are imported into project (but twice). It breaks compilation.

<img width="857" alt="cocoapodsbug" src="https://user-images.githubusercontent.com/11790983/48676666-e8daf000-eb7a-11e8-8cbf-15347aeb402c.png">

Sample project, demonstrated issue:
https://github.com/vojerr/CocoapodsHeadersBug

**Comment by segiddins** [comment]
You'll need to use a `header_mappings_dir` if you want to be able to import both headers, but that doesnt explain why both `.m`s aren't being added correctly

**Comment by dreampiggy** [comment]
I have the same issue.

I'm try to provide a pure C/C++ project, from a standalone Xcode project, into CocoaPods podspec syntax.

It have 3 directory which contains the same name source file `.cpp`, but the content are exact different. I can successfully config and build with Xcode or Carthage.

However, I can not write any correct way to port this into CocoaPods `source_files` syntax.

**Comment by dreampiggy** [comment]
The Xcode project of framework (manage by my own hands) can build success.
![image](https://user-images.githubusercontent.com/6919743/52163383-40b9d980-271c-11e9-844b-fdc6276393fe.png)

**Comment by dreampiggy** [comment]
Try these Podspec syntax. However, the generate project source files list is totally wrong. Which treat any same name source files (for example, `yuv.cpp`), without consider its path. And cause build failsure.

```
s.source_files = 'source/x265.{h,cpp}', 'source/x265_config.h', 'source/common/*.{h,c,cpp,hpp}', 'source/input/*.{h,c,cpp,hpp}', 'source/output/*.{h,c,cpp,hpp}', 'source/encoder/*.{h,c,cpp,hpp}'
```

The generate `Pods.xcproj`:

![image](https://user-images.githubusercontent.com/6919743/52163434-dfded100-271c-11e9-95df-fb682ce9d6d2.png)

**Comment by dreampiggy** [comment]
Right now, I can not find a way to solve this for CocoaPods. I have to pre-built the static library (`.a`) and provide it to my user. It's a litttle suck...

**Comment by dreampiggy** [comment]
The podspec is here. You can use it with `pod 'libx265', :git => 'https://github.com/dreampiggy/x265.git'`

https://github.com/dreampiggy/x265

**Comment by dreampiggy** [comment]
OK. I found a hack way to pass this bug. It's to use Pod subspec to allow same filename exist in one Pod.

And, it take me hours to fix that `pod spec lint` issue, because by default, all the subspecs combination (Mine is 5, the all combinations nearlly 20...) it's hard to pass. So I polish the hack of Podspec to pass. I think that `pod spec lint --help` should point out the `--subspecs=xxx` usage. Or it takes so much of time to find the way out.

The final achievement: https://github.com/SDWebImage/libx265-Xcode/blob/3.0/libx265.podspec#L44-L47

The Pod is now on trunk. The x265 codec: https://cocoapods.org/pods/libx265

**Comment by dreampiggy** [comment]
And, during that Podspec, I found a nother use case of CocoaPods subspecs.

Is there are any way, to make one `subspec`, contains Project Headers only (wihtout any Public Headers or Private Headers ?). I found that there are no way to do this.

I try this workaround, it works in local `:path` usage. But however, this does not pass the `pod spec lint`

```ruby
  s.subspec 'common' do |ss|
    ss.dependency 'libx265/core'
    ss.source_files = 'source/common/*.{h,c,cpp,hpp}'
    ss.public_header_files = 'source/x265.h'# dummy to mark all as project header
  end
```

**Comment by amorde** [comment]
Yes this is a bug, we've marked it as a defect but no one has taken the time to fix it.

PRs are welcomed :)

> Is there are any way, to make one subspec, contains Project Headers only (wihtout any Public Headers or Private Headers ?)

Headers included in `source_files` are `Project` by default - I _think_ this can be overridden on the subspec level but if the root spec includes them in `public_header_files` then you might need to override it as you have done in your example

**Comment by dreampiggy** [comment]
> Yes this is a bug, we've marked it as a defect but no one has taken the time to fix it.
> 
> PRs are welcomed :)
> 
> > Is there are any way, to make one subspec, contains Project Headers only (wihtout any Public Headers or Private Headers ?)
> 
> Headers included in `source_files` are `Project` by default - I _think_ this can be overridden on the subspec level but if the root spec includes them in `public_header_files` then you might need to override it as you have done in your example

I’ll try to provide the solution to fix this. Maybe after the vacation. Though I’m not famaliar with Pods, but I have a little ruby language experience....

**Comment by hasayakey** [comment]
facing this problem.
in my case, i have some c++ file for different platforms with some file's name in the same.

**Comment by Alexufo** [comment]
I have the same problem
